File: stack/stack.c
#include <stdio.h>
#include <stdlib.h>

typedef struct {
    int* array;
    int top;
    int capacity;
} Stack;

Stack* createStack(int capacity) {
    Stack* stack = (Stack*)malloc(sizeof(Stack));
    stack->capacity = capacity;
    stack->top = -1;
    stack->array = (int*)malloc(stack->capacity * sizeof(int));
    return stack;
}

int isEmpty(Stack* stack) {
    return stack->top == -1;
}

int isFull(Stack* stack) {
    return stack->top == stack->capacity - 1;
}

void push(Stack* stack, int data) {
    if (isFull(stack)) {
        printf("Stack is full. Cannot push %d.\n", data);
        return;
    }
    stack->array[++stack->top] = data;
    printf("Pushed %d onto the stack\n", data);
}

void pop(Stack* stack) {
    if (isEmpty(stack)) {
        printf("Stack is empty. Cannot pop.\n");
        return;
    }
    printf("Popped %d from the stack\n", stack->array[stack->top--]);
}

void peek(Stack* stack) {
    if (isEmpty(stack)) {
        printf("Stack is empty. Cannot peek.\n");
        return;
    }
    printf("Top element is %d\n", stack->array[stack->top]);
}

void display(Stack* stack) {
    if (isEmpty(stack)) {
        printf("Stack is empty.\n");
        return;
    }
    printf("Stack elements: ");
    for (int i = 0; i <= stack->top; i++) {
        printf("%d ", stack->array[i]);
    }
    printf("\n");
}

int main() {
    int capacity;
    printf("Enter the capacity of the stack: ");
    scanf("%d", &capacity);

    Stack* stack = createStack(capacity);
    int choice, data;

    while (1) {
        printf("\n1. Push\n2. Pop\n3. Peek\n4. Display\n5. Exit\n");
        printf("Enter your choice: ");
        scanf("%d", &choice);

        switch (choice) {
            case 1:
                printf("Enter data to push: ");
                scanf("%d", &data);
                push(stack, data);
                break;
            case 2:
                pop(stack);
                break;
            case 3:
                peek(stack);
                break;
            case 4:
                display(stack);
                break;
            case 5:
                free(stack->array);
                free(stack);
                exit(0);
            default:
                printf("Invalid choice\n");
        }
    }
    return 0;
}

File: stack/stack.py
class Stack:
    def __init__(self, capacity):
        self.array = [None] * capacity
        self.top = -1
        self.capacity = capacity

    def is_empty(self):
        return self.top == -1

    def is_full(self):
        return self.top == self.capacity - 1

    def push(self, data):
        if self.is_full():
            print(f"Stack is full. Cannot push {data}.")
            return
        self.top += 1
        self.array[self.top] = data
        print(f"Pushed {data} onto the stack")

    def pop(self):
        if self.is_empty():
            print("Stack is empty. Cannot pop.")
            return
        print(f"Popped {self.array[self.top]} from the stack")
        self.top -= 1

    def peek(self):
        if self.is_empty():
            print("Stack is empty. Cannot peek.")
            return
        print(f"Top element is {self.array[self.top]}")

    def display(self):
        if self.is_empty():
            print("Stack is empty.")
            return
        print("Stack elements:", end=" ")
        for i in range(self.top + 1):
            print(self.array[i], end=" ")
        print()

def main():
    capacity = int(input("Enter the capacity of the stack: "))
    stack = Stack(capacity)

    while True:
        print("\n1. Push\n2. Pop\n3. Peek\n4. Display\n5. Exit")
        choice = int(input("Enter your choice: "))

        if choice == 1:
            data = int(input("Enter data to push: "))
            stack.push(data)
        elif choice == 2:
            stack.pop()
        elif choice == 3:
            stack.peek()
        elif choice == 4:
            stack.display()
        elif choice == 5:
            break
        else:
            print("Invalid choice")

if __name__ == "__main__":
    main()
